ref/Microsoft.Build/net/Microsoft.Build.cs

```diff
@@ -1495,6 +1496,19 @@ public partial class ProxyTargets
 }
 namespace Microsoft.Build.FileSystem
 {
+    public delegate bool FindPredicate(ref System.ReadOnlySpan<char> fileName);
+    public delegate TResult FindTransform<TResult>(ref System.ReadOnlySpan<char> fileName);
+    public partial interface IDirectoryCache
+    {
+        bool DirectoryExists(string path);
+        System.Collections.Generic.IEnumerable<TResult> EnumerateDirectories<TResult>(string path, Microsoft.Build.FileSystem.FindPredicate predicate, Microsoft.Build.FileSystem.FindTransform<TResult> transform);
+        System.Collections.Generic.IEnumerable<TResult> EnumerateFiles<TResult>(string path, Microsoft.Build.FileSystem.FindPredicate predicate, Microsoft.Build.FileSystem.FindTransform<TResult> transform);
+        bool FileExists(string path);
+    }
+    public partial interface IDirectoryCacheFactory
+    {
+        Microsoft.Build.FileSystem.IDirectoryCache GetDirectoryCacheForProject(Microsoft.Build.Evaluation.Project project);
```

Comment:
One problem is that we want to keep using the same disk cache for a single project evaluation, so the result can be consistent (if the project has two globs to match the same file, we should match both them together or none to keep it consistent.)  So, maybe we should pass in the evaluationId here?

Reply:
Consistency is controlled by the lifetime of the EvaluationContext instance. When you want msbuild evaluations to read a different state of the disk, you create a new EvaluationContext and start using that one. At least that's the assumption on which the evaluation context code is built on, there's no separation based on the evaluation ID. Also, the evaluation ID isn't propagated to all all the evaluation components, so it's hard for an arbitrary point of evaluation to know what evaluation ID is using it.